src/main.rs

```diff
@@ -83,7 +91,9 @@ fn main() -> io::Result<()> {
         if vm.run().is_err() {
             break;
         }
-        window.update_with_buffer(&vm.buffer, WIDTH, HEIGHT).unwrap();
+        let mut frame_buffer = [0; WIDTH * SCALE * HEIGHT * SCALE];
+        enlarge(&vm.buffer, &mut frame_buffer);
+        window.update_with_buffer(&frame_buffer, WIDTH, HEIGHT).unwrap();
```

Comment:
I think you forgot to add `* SCALE` in function update_with_buffer?
It is fine.
I found that `minifb` actually help handle the enlarge. So you don't need to add enlarge function to enlarge it manually. Just create a larger window with size of `(WIDTH * scale) x (HEIGHT * scale), and use `vm.buffer` as usual.

Reply:
Also, just remind you that your `enlarge` function implementation is incorrect.
Please spend some time to think about it.
As a reference, I prefer to implement like this:
```rust
for (idx, pixel) in frame_buffer.iter_mut().enumerate().rev() {
    let row = idx / (WIDTH * scale * scale);
    let col = (idx % (WIDTH * scale)) / scale;
    *pixel = buffer[row * WIDTH + col];
    // or you can make it one line version but less readable.
}
```
it contains less loop, and it applies to the situation that buffer and frame_buffer are same buffers.

Reply:
Yes, I realize in this morning that I never implement this function correctly from the very beginning. The only thing I relied  on is minifb's auto render.